app/shared/models/git_repo.rb

```diff
@@ -106,32 +118,55 @@ def initialize(git_config: nil, local_folder: nil, provider_credential: nil, asy
         raise "unsupported credential type: #{provider_credential.type}"
       end
 
-      logger.debug("Adding task to setup repo #{self.git_config.git_url} at: #{local_folder}")
+      logger.debug("Adding task to setup repo #{git_config.git_url} at: #{local_folder}")
 
       setup_task = git_action_with_queue(ensure_block: proc { callback_block(async_start) }) do
-        logger.debug("Starting setup_repo #{self.git_config.git_url}".freeze)
+        logger.debug("Starting setup_repo #{git_config.git_url}".freeze)
         setup_repo
-        logger.debug("Done setup_repo #{self.git_config.git_url}".freeze)
+        logger.debug("Done setup_repo #{git_config.git_url}".freeze)
       end
 
       # if we're starting asynchronously, we can return now.
       if async_start
-        logger.debug("Asynchronously starting up repo: #{self.git_config.git_url}")
+        logger.debug("Asynchronously starting up repo: #{git_config.git_url}")
         return
       end
 
-      logger.debug("Synchronously starting up repo: #{self.git_config.git_url} at: #{local_folder}")
+      logger.debug("Synchronously starting up repo: #{git_config.git_url} at: #{local_folder}")
       now = Time.now.utc
       sleep_timeout = now + sync_setup_timeout_seconds # 10 second startup timeout
       while !setup_task.completed && now < sleep_timeout
         time_left = sleep_timeout - now
-        logger.debug("Not setup yet, sleeping (time before timeout: #{time_left}) #{self.git_config.git_url}")
+        logger.debug("Not setup yet, sleeping (time before timeout: #{time_left}) #{git_config.git_url}")
         sleep(2)
         now = Time.now.utc
       end
 
       raise "Unable to start git repo #{git_config.git_url} in #{sync_setup_timeout_seconds} seconds" if now > sleep_timeout
-      logger.debug("Done starting up repo: #{self.git_config.git_url}")
+      logger.debug("Done starting up repo: #{git_config.git_url}")
+    end
+
+    def handle_exception(ex, message: nil)
```

Comment:
Would be good to have documentation here on why we add `message` as a separate parameter